storage-node/src/services/queryNode/api.ts

```diff
@@ -285,11 +217,11 @@ export class QueryNodeApi {
    * @param bucketIds - query filter: bucket IDs
    */
   public async getStorageBagsDetails(bucketIds: string[]): Promise<Array<StorageBagDetailsFragment>> {
-    const result = await this.multipleEntitiesWithPagination<
-      StorageBagDetailsFragment,
-      GetBagConnectionQuery,
-      GetBagConnectionQueryVariables
-    >(GetBagConnection, { limit: MAX_RESULTS_PER_QUERY, bucketIds }, 'storageBagsConnection')
+    const result = await this.multipleEntitiesQuery<GetStorageBagDetailsQuery, GetStorageBagDetailsQueryVariables>(
```

Comment:
Why are we moving away from paginated query here?

Reply:
I think the pagination was mainly used due to the limitation of the QN, and storage squid does not have this limitation and returns the result successfully. In fact, having pagination adds overhead, both network (multiple round-trips) as well as DB, since db has to execute the same query each time with a different offset.

Reply:
Just a comment on generators. I was experimenting here https://github.com/Joystream/joystream/pull/5026/files#diff-0da8a15d4ae4a081c01cccf049103d2c94906628d9d0d57b715c991d3f2596bd with using generator to yield a page at a time to avoid having to concat results into very large arrays.

Until we persist some state there is always going to be this need when colossus starts up to refetch "everything".

And as Zeeshan is suggesting, we both recognize that it make more sense after this initial fetch everything, subsequent sync runs only need to look at the "diff", what new objects have been created since last sync run (in addition to what needs to be re-tried from last run)